src/Runner.Worker/Container/ContainerHooks/ContainerHookManager.cs

```diff
@@ -52,37 +52,40 @@ public async Task PrepareJobAsync(IExecutionContext context, List<ContainerInfo>
 
             var response = await ExecuteHookScript(context, input);
             // TODO: Should we throw if response.Context is null or just noop?
-            context.JobContext.Container["id"] = new StringContextData(response.Context.Container.Id);
-            jobContainer.ContainerId = response.Context.Container.Id;
-            context.JobContext.Container["network"] = new StringContextData(response.Context.Container.Network);
-            jobContainer.ContainerNetwork = response.Context.Container.Network;
-            context.JobContext["hook_state"] = new StringContextData(JsonUtility.ToString(response.State));
-
-            // TODO: figure out if we need ContainerRuntimePath for anything
-            // var configEnvFormat = "--format \"{{range .Config.Env}}{{println .}}{{end}}\"";
-            // var containerEnv = await _dockerManager.DockerInspect(executionContext, container.ContainerId, configEnvFormat);
-            // container.ContainerRuntimePath = DockerUtil.ParsePathFromConfigEnv(containerEnv);
-
-            for (var i = 0; i < response.Context.Services.Count; i++)
+            if (response.Context != null)
             {
-                var container = response.Context.Services[i]; // TODO: Confirm that the order response.Context.Services is the same as serviceContainers
-                var containerInfo = serviceContainers[i];
-                containerInfo.ContainerId = container.Id;
-                containerInfo.ContainerNetwork = container.Network;
-                var service = new DictionaryContextData()
+                context.JobContext.Container["id"] = new StringContextData(response.Context.Container.Id);
```

Comment:
any of these fields could be null even if context is not null.

I think we will have to do checks for each field, 

```
if ( response.Context.Container)
{
  if (response.Context.Container.Id)
  {
     context.JobContext.Container["id"] = new StringContextData(response.Context.Container.Id);
  }
    if (response.Context.Container.Network)
   {
// ...
   } 
}
```